pkg/engine/planner/logical/logical_test.go

```diff
@@ -0,0 +1,58 @@
+package logical
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestPlan_String(t *testing.T) {
+	// Build a simple query plan:
+	// SELECT id, name FROM users WHERE age > 21 SORT BY age ASC
+	df := NewDataFrame(
+		&MakeTable{
+			Selector: &BinOp{
+				Left:  &ColumnRef{Column: "table", Type: ColumnTypeLabel},
+				Right: LiteralString("users"),
+				Op:    BinOpKindEq,
+			},
+		},
+	).Select(
+		&BinOp{
+			Left:  &ColumnRef{Column: "age", Type: ColumnTypeMetadata},
+			Right: LiteralInt64(21),
+			Op:    BinOpKindGt,
+		},
+	).Sort(ColumnRef{Column: "age", Type: ColumnTypeMetadata}, true, false)
+
+	// Convert to SSA
+	ssaForm, err := df.ToPlan()
+	require.NoError(t, err)
+	require.NotNil(t, ssaForm)
+
+	t.Logf("SSA Form:\n%s", ssaForm.String())
+
+	// Define expected output
+	exp := `
+%1 = EQ label.table, "users" 
+%2 = MAKE_TABLE [selector=%1] 
```

Comment:
I'm not in love with the SSA output for the selector (what is the type of `%1`? Some kind of closure?) but I don't want to spend too much time messing around with it; it feels like something we can clean up later.

Reply:
I don't think that we have to worry about the representation too much (at least for now). The SSA is more or less an internal that is never directly exposed to users anyway.